src/components/playback/playbackmanager.js

```diff
@@ -1316,16 +1317,20 @@ class PlaybackManager {
 
             return profiles.filter(function (p) {
                 if (p.Type === 'Video') {
-                    if (!p.AudioCodec) {
-                        return true;
-                    }
+                    const containers = (p.Container || '').toLowerCase().split(',');
 
-                    // This is an exclusion filter
-                    if (p.AudioCodec.indexOf('-') === 0) {
-                        return p.AudioCodec.toLowerCase().indexOf(codec) === -1;
-                    }
+                    if (containers.includes(container)) {
+                        if (!p.AudioCodec) {
+                            return true;
+                        }
 
-                    return p.AudioCodec.toLowerCase().indexOf(codec) !== -1;
+                        // This is an exclusion filter
+                        if (p.AudioCodec.startsWith('-')) {
+                            return !p.AudioCodec.substring(1).toLowerCase().split(',').includes(codec);
+                        }
```

Comment:
Should I also add an exclusive filter to the player?
Or just remove it from everywhere?

Reply:
Since the server [supports an exclusive list](https://github.com/jellyfin/jellyfin/blob/c795f17fa6ae9e4f68f23cb569a1129de8fd7635/MediaBrowser.Model/Dlna/ContainerProfile.cs#L38-L43), we need to add it for the player as well.